src/clr-angular/data/datagrid/providers/column-reorder.service.spec.ts

```diff
@@ -0,0 +1,180 @@
+    let fixture: ComponentFixture<any>;
+
+    let testComponent: TestComponent;
+    let columnReorderService: ColumnReorderService;
+    let columnsService: MockColumnsService;
+
+    beforeEach(function() {
+      TestBed.configureTestingModule({
+        declarations: [TestComponent],
+        providers: [MOCK_COLUMN_SERVICE_PROVIDER, ColumnReorderService],
+      });
+      fixture = TestBed.createComponent(TestComponent);
+      fixture.detectChanges();
+      columnReorderService = TestBed.get(ColumnReorderService);
+      columnsService = <MockColumnsService>TestBed.get(ColumnsService);
+      columnsService.mockColumns(3);
+
+      testComponent = fixture.componentInstance;
+    });
+
+    afterEach(() => {
+      fixture.destroy();
+    });
+
+    it('provides the unique string id for column group', function() {
+      expect(columnReorderService.columnsGroupId).toBeString();
+    });
+
+    it('provides the view container', function() {
+      expect(columnReorderService.containerRef).not.toBeUndefined();
+    });
+
+    it('emits reorder request data when reorderViews method gets called', function() {
+      expect(testComponent.reorderRequest).toBeUndefined();
+      columnReorderService.reorderViews(testComponent.view1, testComponent.view2, null);
+      expect(testComponent.reorderRequest).not.toBeUndefined();
+    });
+
+    it('updates orders property value through updateOrders method', function() {
+      expect(columnReorderService.orders).toBeUndefined();
+      columnReorderService.updateOrders([1, 0, 2]);
+      expect(columnReorderService.orders).toEqual([1, 0, 2]);
+    });
+
+    it('updates column states on orders', function() {
+      expect(columnsService.columns[0].value.order).toBeUndefined();
+      expect(columnsService.columns[1].value.order).toBeUndefined();
+      expect(columnsService.columns[2].value.order).toBeUndefined();
+
+      columnReorderService.updateOrders([1, 0, 2]);
+
+      expect(columnsService.columns[0].value.order).toBe(1);
+      expect(columnsService.columns[1].value.order).toBe(0);
+      expect(columnsService.columns[2].value.order).toBe(2);
+    });
+
+    it('requests first and last visible columns checks after updating orders', function() {
+      expect(testComponent.firstVisibleChecked).toBeFalsy();
+      expect(testComponent.lastVisibleChecked).toBeFalsy();
+      columnReorderService.updateOrders([1, 0, 2]);
+      expect(testComponent.firstVisibleChecked).toBeTruthy();
+      expect(testComponent.lastVisibleChecked).toBeTruthy();
+    });
+
+    it('returns new order at index', function() {
+      columnReorderService.orders = [1, 0, 2];
+      expect(columnReorderService.orderAt(0)).toBe(1);
+      expect(columnReorderService.orderAt(1)).toBe(0);
+      expect(columnReorderService.orderAt(2)).toBe(2);
+      expect(columnReorderService.orderAt(5)).toBe(-1);
+    });
+
+    describe('should emit proper reorder request data on moving:', () => {
```

Comment:
nit: can we keep to either arrow functions or a standard function declarations?

some describes/its use:
```typescript
describe('my suite', function() {...})
```
other describes/its use:
```typescript
describe('should emit proper reorder request data on moving:', () => {...}
```

Reply:
definitely